packages/blocks/src/page-block/edgeless/components/rects/edgeless-selected-rect.ts

```diff
@@ -621,7 +624,7 @@ export class EdgelessSelectedRect extends WithDisposable(LitElement) {
     this.slots.cursorUpdated.emit(cursor);
   };
 
-  private _updateSelectedRect() {
+  private _updateSelectedRect = withRequestConnectedFrame(() => {
```

Comment:
Is this equivalent of plain `requestAnimationFrame`? If so, don't encapsulation opaque utils since I haven't seem the chance reusing it elsewhere.

Reply:
It's not equivalent of `requestAnimationFrame`.
It ensures that the function will be executed at most once per frame, no matter how many times it is called.

<img width="626" alt="image" src="https://github.com/toeverything/blocksuite/assets/58546692/1bb28cb1-31c9-4a0f-9148-03b3c5fe14d5">
<img width="851" alt="image" src="https://github.com/toeverything/blocksuite/assets/58546692/5e28c79e-41f8-4ef9-9e23-afb7714ec59f">

I see some patterns here. I can replace them.